src/components/Nav/Nav.js

```diff
@@ -1,85 +1,102 @@
-import React, { Component } from 'react'
-import './Nav.scss'
-import { NavLink, withRouter } from 'react-router-dom'
-import { connect } from 'react-redux'
-import { PropTypes } from 'prop-types'
-import { loginUser, setErrorMessage, storeFavorites } from '../../actions'
-import allPurposeFetch from '../../allPurposeFetch';
+import React, { Component } from "react";
+import "./Nav.scss";
+import { NavLink, withRouter } from "react-router-dom";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+import { loginUser, setErrorMessage, storeFavorites } from "../../actions";
+import allPurposeFetch from "../../allPurposeFetch";
 
-const Nav = (props) => {
+const Nav = props => {
+  console.log(props)
+  const signOut = e => {
+    const data = {
+      name: "",
+      email: ""
+    };
+    props.loginUser(data);
+  };
 
-    const signOut = (e) => {
-        const data = {
-            name: '',
-            email: '',
-        }
-        props.loginUser(data)
+  const validateLogin = async e => {
+    const { history } = props;
+    if (!props.user.email) {
+      props.setErrorMessage("You must be signed in to view favorites");
```

Comment:
I think we'll want to move these error messages to the store, but we'll get to that next.